lua/lualine/component.lua

```diff
@@ -105,7 +105,13 @@ function M:apply_padding()
     end
   end
   if r_padding then
-    self.status = self.status .. string.rep(' ', r_padding)
+    if self.status:reverse():find('%*%%.*#.*#%%') == 1 then
+      -- When component has changed the highlight at the end
+      -- we will add the padding before the highlight terminates
+      self.status = self.status:sub(1, -3) .. string.rep(' ', r_padding) .. self.status:sub(-2, -1)
```

Comment:
Explain this line please.  Why are you removing only last 2 chars and not the HL group mapping %#group_name# ?

Reply:
This line inserts padding at the _end_ of the `<text>` part of a string of the form `%#Group#<text>%*`, so the only characters to hold out are the last two characters `%*`.